controller/features.go

```diff
@@ -2,29 +2,103 @@ package controller
 
 import (
 	"fmt"
+	jwtgo "github.com/dgrijalva/jwt-go"
 	"github.com/fabric8-services/fabric8-toggles-service/app"
+	"github.com/fabric8-services/fabric8-toggles-service/configuration"
+	"github.com/fabric8-services/fabric8-toggles-service/featuretoggles"
+	"github.com/fabric8-services/fabric8-wit/log"
 	"github.com/goadesign/goa"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
 	uuid "github.com/satori/go.uuid"
+	"strings"
 )
 
 // FeaturesController implements the features resource.
 type FeaturesController struct {
 	*goa.Controller
+	config *configuration.Data
 }
 
 // NewFeaturesController creates a features controller.
-func NewFeaturesController(service *goa.Service) *FeaturesController {
-	return &FeaturesController{Controller: service.NewController("FeaturesController")}
+func NewFeaturesController(service *goa.Service, config *configuration.Data) *FeaturesController {
+	return &FeaturesController{
+		Controller: service.NewController("FeaturesController"),
+		config:     config,
+	}
 }
 
 // List runs the list action.
 func (c *FeaturesController) List(ctx *app.ListFeaturesContext) error {
 	// FeaturesController_List: start_implement
 
-	// Put your logic here
+	jwtToken := goajwt.ContextJWT(ctx)
+	groupId := ""
+	if jwtToken != nil {
+		groupId = jwtToken.Claims.(jwtgo.MapClaims)["company"].(string) // TODO replace with a custom claim group_id
+	}
+	var enableFeatures *app.FeatureList
+	if c.config != nil {
+		featuresListFromUnleash, err := getFeatureListFromUnleashServer(ctx, c.config, groupId)
```

Comment:
for your tests, you could define an `unleashService` that has its own `unleashClient`, then replace that client with a mock, so you can do whatever you want without depending on an actual deployed unleash server